Validation step fails with SyntaxError: Unexpected number in JSON when using is-vaild-extenstion-agent-build-task.IsAzureDevopsextensionValid@4

Summary
Attempting to add a post-publish validation step with the task
ms-devlabs.vsts-developer-tools-build-tasks.is-vaild-extenstion-agent-build-task.IsAzureDevopsextensionValid@4
fails with a JSON parse error even though the CLI reports status: "success".
YAML used

    `- task: ms-devlabs.vsts-developer-tools-build-tasks.is-vaild-extenstion-agent-build-task.IsAzureDevopsextensionValid@4 
      displayName: 'Validate extension using extension ID and version'
      inputs:
        connectedServiceName: $(marketplace_serviceconnection)
        publisherId: $(marketplace_publisherId)
        extensionId: $(marketplace_extensionId)
        extensionVersion: $(extensionVersion)
        maxAttempts: 40
        attemptDelay: 300`


Observed logs

 /azp/agent/_work/_tasks/npm_local/node_modules/.bin/tfx extension isvalid -json ...
loader.load
}
"status" : "success"
2025-10-27T13:01:02.812Z : loader.load
##[error]Extension validation failed: Sytax Error: Unexpected number in Json at position 4

--- a/BuildTasks/IsValidExtensionAgent/v4/IsValidExtension.ts
+++ b/BuildTasks/IsValidExtensionAgent/v4/IsValidExtension.ts
@@ -21,10 +21,15 @@ async function run() {
             };
 
             await promiseRetry(options,
-                (retry, attempt) => {
+                async (retry, attempt) => {
                     tl.debug(`Attempt: ${attempt}`);
-                    const result = tfx.execSync({ silent: false, failOnStdErr: false } as tlr.IExecSyncOptions);
-                    const json = JSON.parse(result.stdout);
+                    
+                    const outputStream = new common.TfxJsonOutputStream(console.log);
+                    const errorStream = new common.TfxJsonOutputStream(tl.error);
+                    
+                    await tfx.execAsync({ outStream: outputStream, errorStream: errorStream, failOnStdErr: false, ignoreReturnCode: true } as tlr.IExecOptions);
+                    
+                    const json = JSON.parse(outputStream.jsonString);
                     switch (json.status) {
                         case "pending":
                             return retry(json.status);

--- a/BuildTasks/IsValidExtensionAgent/v5/IsValidExtension.ts
+++ b/BuildTasks/IsValidExtensionAgent/v5/IsValidExtension.ts
@@ -1,4 +1,5 @@
 import tl from "azure-pipelines-task-lib";
+import tr from "azure-pipelines-task-lib/toolrunner.js";
 import * as common from "../../Common/v5/Common.js";
 import promiseRetry from "promise-retry";
 
@@ -18,10 +19,15 @@ await common.runTfx(async tfx => {
         };
 
         await promiseRetry(options,
-            (retry, attempt) => {
+            async (retry, attempt) => {
                 tl.debug(`Attempt: ${attempt}`);
-                const result = tfx.execSync({ silent: false, failOnStdErr: false } as any);
-                const json = JSON.parse(result.stdout);
+                
+                const outputStream = new common.TfxJsonOutputStream(console.log);
+                const errorStream = new common.TfxJsonOutputStream(tl.error);
+                
+                await tfx.execAsync({ outStream: outputStream, errorStream: errorStream, failOnStdErr: false, ignoreReturnCode: true } as tr.IExecOptions);
+                
+                const json = JSON.parse(outputStream.jsonString);
                 switch (json.status) {
                     case "pending":
                         return retry(json.status);

--- a/test-isvalid.cmd
+++ b/test-isvalid.cmd
@@ -1,0 +1,61 @@
+set INPUT_METHOD=id
+set INPUT_PUBLISHERID=jessehouwing
+set INPUT_EXTENSIONID=jessehouwing-vsts-variable-tasks-dev
+set INPUT_EXTENSIONVERSION=3.0.1122
+SET INPUT_CONNECTEDSERVICENAME=A
+SET INPUT_VERSION=builtin
+set azdo_token=
+SET ENDPOINT_URL_A=https://marketplace.visualstudio.com
+SET ENDPOINT_AUTH_A={ "parameters": { "password": "%azdo_token%" }, "Scheme": "basic" }
+
+set AGENT_WORKFOLDER=%temp%\agent\work
+set AGENT_TOOLSDIRECTORY=%temp%\agent\tools
+SET AGENT_TEMPDIRECTORY=%temp%\agent\tmp
+set __TFXPATH=E:\azure-devops-extension-tasks\BuildTasks\TfxInstaller\v5\node_modules\.bin
+md %temp%\agent
+md %AGENT_WORKFOLDER%
+md %AGENT_TOOLSDIRECTORY%
+md %AGENT_TEMPDIRECTORY%
+
+set NODE_ENV=production
+set NO_UPDATE_NOTIFIER=true
+
+REM cmd /c "npm run build:tasks"
+
+pushd BuildTasks\TfxInstaller\v5\TfxInstaller\v5
+node TfxInstaller.js
+popd
+
+echo.
+echo ========================================
+echo Testing IsValidExtension v4
+echo ========================================
+pushd BuildTasks\IsValidExtensionAgent\v4\IsValidExtensionAgent\v4
+node IsValidExtension.js
+popd
+
+echo.
+echo ========================================
+echo Testing IsValidExtension v5
+echo ========================================
+pushd BuildTasks\IsValidExtensionAgent\v5\IsValidExtensionAgent\v5
+node IsValidExtension.js
+popd
+
+echo.
+echo ========================================
+echo Testing IsValidExtension v4 with TFX_TRACE=1
+echo ========================================
+set TFX_TRACE=1
+pushd BuildTasks\IsValidExtensionAgent\v4\IsValidExtensionAgent\v4
+node IsValidExtension.js
+popd
+
+echo.
+echo ========================================
+echo Testing IsValidExtension v5 with TFX_TRACE=1
+echo ========================================
+pushd BuildTasks\IsValidExtensionAgent\v5\IsValidExtensionAgent\v5
+node IsValidExtension.js
+popd
+set TFX_TRACE=

